commons/com.b2international.index/src/com/b2international/index/es/EsDocumentSearcher.java

```diff
@@ -380,9 +382,13 @@ private String toSearchAfterToken(final Object[] searchAfter) {
 		
 		try (final DataInputStream dis = new DataInputStream(new ByteArrayInputStream(decodedToken))) {
 			JavaBinCodec codec = new JavaBinCodec();
-			List<?> obj = (List<?>) codec.unmarshal(dis);
-			codec.close();
-			return obj.toArray();
+			try {
+				List<?> obj = (List<?>) codec.unmarshal(dis);
+				codec.close();
+				return obj.toArray();
+			} catch (IllegalArgumentException e) {
+				throw new BadRequestException(e.getMessage(), e.getCause());
```

Comment:
Use a more appropriate message than the internal `the data in not in 'javabin' format`. Like
```suggestion
				throw new BadRequestException("Invalid 'searchAfter' parameter value '%s'", searchAfterToken);
```